com.reprezen.swagedit/src/com/reprezen/swagedit/editor/outline/SwaggerContentOutlinePage.java

```diff
@@ -0,0 +1,66 @@
+package com.reprezen.swagedit.editor.outline;
+
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.viewers.SelectionChangedEvent;
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.ui.part.IShowInTarget;
+import org.eclipse.ui.part.ShowInContext;
+import org.eclipse.ui.texteditor.IDocumentProvider;
+import org.eclipse.ui.views.contentoutline.ContentOutlinePage;
+
+import com.reprezen.swagedit.editor.SwaggerDocument;
+import com.reprezen.swagedit.model.Model;
+
+public class SwaggerContentOutlinePage extends ContentOutlinePage {
+
+    private final IDocumentProvider documentProvider;
+    private final IShowInTarget showInTarget;
+
+    private Object currentInput;
+
+    public SwaggerContentOutlinePage(IDocumentProvider documentProvider, IShowInTarget showInTarget) {
+        super();
+        this.documentProvider = documentProvider;
+        this.showInTarget = showInTarget;
+    }
+
+    public void setInput(Object input) {
+        this.currentInput = input;
+        update();
+    }
+
+    @Override
+    public void createControl(Composite parent) {
+        super.createControl(parent);
+
+        TreeViewer viewer = getTreeViewer();
+        viewer.setContentProvider(new OutlineContentProvider());
+        viewer.setLabelProvider(new OutlineStyledLabelProvider());
+        viewer.addSelectionChangedListener(this);
+        viewer.setAutoExpandLevel(2);
```

Comment:
I found a bug which may be related to `setAutoExpandLevel(2)`
I noticed that if I expand an element in the Outline View, then edit the element in the editor, the corresponding node in the outline which was previously expanded gets collapsed:
![outline_collapse](https://cloud.githubusercontent.com/assets/644582/17605090/b0413014-5fe6-11e6-95fe-df6f0b7619b4.gif)


It's a bug.